tutorials/juliacon2022.jl

```diff
@@ -0,0 +1,1644 @@
+
+# ╔═╡ c78b8938-2f4e-4b21-afa0-01c6aa03b689
+for f in files
+	localpath = joinpath(".", f)
+	isfile(localpath) && continue # skip file if it's already downloaded
+	Downloads.download(joinpath(baseurl, f), localpath)
+end
+
+# ╔═╡ 64624649-8a70-44fb-bbd5-5c3466ce484a
+@assert all(isfile, files)
+
+# ╔═╡ ce747551-460b-4542-b07c-7adc3fdd08cd
+md"""
+### 📊 Data format
+
+The data we've downloade here is in the form of tab-separated files,
+one per biosample,
+where rows represent individual features - in this case, indivudual taxa.
+
+Let's take a look:
+"""
+
+# ╔═╡ 8a45213f-2429-4035-a782-81e5bf8fba17
+first(eachline("SRS014459-Stool_profile.txt"), 6)
+
+# ╔═╡ 37940aa2-2876-4d3a-bed4-306aa42d5f42
+md"""
+The first few rows are headers showing how the file was generated,
+the actual data starts on line 4.
+The first and second columns contain the taxonomy of the feature
+(in human readable form and in NCBI's ID)
+and the third column
+is the relative abundance of that taxon.
+The final column contains information about any additional species
+that may be contained in a given identified row,
+but we can ignore that for now.
+
+Let's load it into a DataFrame to start with:
+"""
+
+# ╔═╡ 370e4f43-3835-45d1-b614-5732c66842f5
+stool = CSV.read("SRS014459-Stool_profile.txt", DataFrame;
+	header = 4, # get the column headers from the 4th line
+	skipto = 5, # start parsing on the 5th line
+	delim = '\t', # use tabs 
+)
+
+# ╔═╡ 88564be4-7943-4577-adee-b5e96351912b
+md"""
+## The `CommunityProfile` type
+
+The main contribution of `Microbiome.jl` is the `CommunityProfile` type,
+which consists of 3 parts:
+
+1. A sparse matrix containing the relative abundances of features.
+2. A list of features. These index the rows of the matrix
+3. A list of biosamples with optionally attached metadata
+   (see `MicrobiomeSample` below). These index the columns of the matrix
+
+We'll contruct it mannually first,
+and then later we'll see how we can use the `Tables.jl` integration
+to our advantage.
+But first, let's load in all of our samples:
+"""
+
+# ╔═╡ 4a5cfe7c-109f-4bb0-ba17-6a4aba9dc16a
+comm_df = let dfs = DataFrame[]
+	for f in files
+		df = CSV.read(f, DataFrame;
+			header = 4, # get the column headers from the 4th line
+			skipto = 5, # start parsing on the 5th line
+			delim = '\t', # use tabs 
+		)
+	
+		samplename=splitext(f)[1] # get sample name from file
+		
+		rename!(df, "#clade_name" => "taxon", "relative_abundance" => samplename)
+		push!(dfs, select(df, [1,3]))
+	end
+	outerjoin(dfs...; on="taxon")
+end
+
+# ╔═╡ 287fd231-4ff9-4c2e-a55d-1786a7e41659
+md"""
+### 🏁 Relative abundance matrix
+
+You can use something from the `SparseArrays` standard library,
+but any old matrix will do, and `Microbiome.jl` will convert it for you.
+
+Notice that, in the DataFrame above,
+a number of taxa are missing from a number of samples.
+This is expected,
+but "missing" in the contex of microbial community data just means
+that the relative abundance is 0.
```

Comment:
```suggestion
that the relative abundance is 0 (zero).
```

in Pluto formatting/font, the 0 is not very clearly a 0. maybe put the actual word zero in parentheses for clarity?